lightning_examples/finetuning-scheduler/requirements.txt

```
@@ -1,6 +1,3 @@
-lightning  # todo: the tuner depends on L so later using with PL crash
-finetuning-scheduler[examples] ==2.3.*
-datasets ==2.17.*
-# todo: pin version intill reinstall with PT-eco alignment
-torch ==2.1.*
-torchvision ==0.16.*
+datasets >=2.17.0  # to allow explicitly setting `trust_remote_code`
+lightning <=2.4.0  # todo: the tuner depends on L so later using with PL crash
```

```suggestion
lightning <2.4.0  # todo: the tuner depends on L so later using with PL crash
```
also, would it be possible to make it work also with PL, let's see for example
https://github.com/Lightning-AI/lightning-Habana/blob/2fd33c6512155a1504df7d13b637ba3b0edaff8e/src/lightning_habana/utils/hpu_distributed.py#L19-L26

---

also, would it be possible to make it work also with PL, let's see for example
https://github.com/Lightning-AI/lightning-Habana/blob/2fd33c6512155a1504df7d13b637ba3b0edaff8e/src/lightning_habana/utils/hpu_distributed.py#L19-L26

---

I've tested locally with the `2.4.0` stack of lightning, pytorch and FTS and was thinking we could hold merging this until the Lightning and FTS official `2.4.x` releases are issued if you want to (either way, I'll re-run the tutorial when Lightning and FTS `2.4.0`  are out to verify everything looks good).

---

I'll take a look at making the example L vs PL agnostic as you suggest and push a commit, should be possible (FTS can be used with either installation mode)

---

The reason is that it moved all L to PL, but then it complains about inheritance even though the code is identical. Regarding the 2.4, it is great it is compatible but until it is tested I would leave it with the latest released, and now the dependable we will have bumps tested as you can see for example #329 #345 #347